wandb/sdk/wandb_init.py

```diff
@@ -590,19 +590,15 @@ def init(self) -> Union[Run, RunDisabled, None]:  # noqa: C901
                 tel.env.kaggle = True
             if self.settings._windows:
                 tel.env.windows = True
-            run._telemetry_imports(tel.imports_init)
 
             if self.settings.launch:
                 tel.feature.launch = True
 
-            if active_start_method == "spawn":
-                tel.env.start_spawn = True
-            elif active_start_method == "fork":
-                tel.env.start_fork = True
-            elif active_start_method == "forkserver":
-                tel.env.start_forkserver = True
-            elif active_start_method == "thread":
-                tel.env.start_thread = True
+            for module_name in telemetry.list_telemetry_imports(only_imported=True):
+                setattr(tel.imports_init, module_name, True)
+
+            if active_start_method in ["spawn", "fork", "forkserver", "thread"]:
+                setattr(tel.env, f"start_{active_start_method}", True)
```

Comment:
old code was type checked, this one isnt... i prefer the old way

Reply:
hmm...i can revert this change what do you mean by type-checked the explicit string comparison?

Reply:
if you do a setattr of something that isnt in the object, there is no static checking..

when you do:
tel.env.start_spawn = True

then mypy will run its static checker and alert you if you are doing something wrong.
We should only use the setattr trick if it is necessary... a huge list of things that isnt convenient to enumerate in code.

alternatively, we can generate a function which takes a bunch of literals like done with:
```
wandb/sdk/lib/_wburls_generated.py
```

Reply:
oh i see what you saying now... hmmm... ok... i'll revert, but lets do the auto generation going forward